test/Microsoft.AspNetCore.Mvc.RazorPages.Test/Internal/PageSaveTempDataPropertyFilterTest.cs

```diff
@@ -0,0 +1,288 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc.Abstractions;
+using Microsoft.AspNetCore.Mvc.Filters;
+using Microsoft.AspNetCore.Mvc.ModelBinding;
+using Microsoft.AspNetCore.Mvc.Rendering;
+using Microsoft.AspNetCore.Mvc.ViewFeatures;
+using Microsoft.AspNetCore.Mvc.ViewFeatures.Internal;
+using Microsoft.AspNetCore.Routing;
+using Moq;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Mvc.RazorPages.Internal
+{
+    public class PageSaveTempDataPropertyFilterTest
+    {
+        [Fact]
+        public void OnTempDataSaving_PopulatesTempDataWithNewValuesFromPageProperties()
+        {
+            // Arrange
+            var httpContext = new DefaultHttpContext();
+
+            var tempData = new TempDataDictionary(httpContext, Mock.Of<ITempDataProvider>());
+
+            var page = new TestPage()
+            {
+                Test = "TestString",
+                Test2 = "Test2",
+            };
+
+            var filter = CreatePageSaveTempDataPropertyFilter(tempData);
+            filter.Subject = page;
+
+            var pageType = page.GetType();
+
+            var testProperty = pageType.GetProperty(nameof(TestPage.Test));
+            var test2Property = pageType.GetProperty(nameof(TestPage.Test2));
+
+            filter.OriginalValues[testProperty] = "SomeValue";
+            filter.OriginalValues[test2Property] = "Test2";
+
+            filter.Properties = new List<TempDataProperty>
+            {
+                new TempDataProperty("TempDataProperty-Test", testProperty, testProperty.GetValue, testProperty.SetValue),
+                new TempDataProperty("TempDataProperty-Test2", test2Property, test2Property.GetValue, test2Property.SetValue)
+            };
+
+            // Act
+            filter.OnTempDataSaving(tempData);
+
+            // Assert
+            Assert.Equal("TestString", page.Test);
+            Assert.Equal("TestString", tempData["TempDataProperty-Test"]);
```

Comment:
Can you add an assert that says `TempDataProperty-Test2` isn't stashed in the tdd?